app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/General.php

```diff
@@ -41,20 +42,27 @@ class General extends AbstractModifier
      */
     private $attributeRepository;
 
+    protected $isProductQtyDecimal;
+
     /**
      * @param LocatorInterface $locator
      * @param ArrayManager $arrayManager
      * @param AttributeRepositoryInterface|null $attributeRepository
+     * @param ProductStockIsQtyDecimal $isProductQtyDecimal
+     *
+     * @return void
      */
     public function __construct(
         LocatorInterface $locator,
         ArrayManager $arrayManager,
-        AttributeRepositoryInterface $attributeRepository = null
+        AttributeRepositoryInterface $attributeRepository = null,
+        ProductStockIsQtyDecimal $isProductQtyDecimal
     ) {
         $this->locator = $locator;
         $this->arrayManager = $arrayManager;
         $this->attributeRepository = $attributeRepository
             ?: \Magento\Framework\App\ObjectManager::getInstance()->get(AttributeRepositoryInterface::class);
+        $this->isProductQtyDecimal = $isProductQtyDecimal;
```

Comment:
It makes sense either to add new dependency in a Backwards Compatible way or refactor the one above to be BiC. Please align these.